lib/node_modules/@stdlib/blas/base/cscal/README.md

```diff
@@ -0,0 +1,242 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2024 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# cscal
+
+> Scales a single-precision complex floating-point vector by a single-precision complex floating-point constant.
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var cscal = require( '@stdlib/blas/base/cscal' );
+```
+
+#### cscal( N, ca, cx, strideX )
+
+Scales values from `cx` by `ca`.
+
+```javascript
+var Complex64Array = require( '@stdlib/array/complex64' );
+var Complex64 = require( '@stdlib/complex/float32' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+
+var cx = new Complex64Array( [ 1.0, 1.0, 1.0, 1.0, 1.0, 1.0 ] );
+var ca = new Complex64( 2.0, 0.0 );
+
+cscal( 3, ca, cx, 1 );
+
+var z = cx.get( 0 );
+// returns <Complex64>
+
+var re = realf( z );
+// returns 2.0
+
+var im = imagf( z );
+// returns 2.0
+```
+
+The function has the following parameters:
+
+-   **N**: number of indexed elements.
+-   **ca**: scalar [`Complex64`][@stdlib/complex/float32] constant. 
+-   **cx**: input [`Complex64Array`][@stdlib/array/complex64].
+-   **strideX**: index increment for `x`.
+
+The `N` and stride parameters determine how values from `cx` are scaled by `ca`. For example, to scale every other value in `cx` by `ca`,
+
+```javascript
+var Complex64Array = require( '@stdlib/array/complex64' );
+var Complex64 = require( '@stdlib/complex/float32' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+
+var cx = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 ] );
+var ca = new Complex64( 2.0, 0.0 );
+
+cscal( 2, ca, cx, 2 );
+
+var z = cx.get( 2 );
+// returns <Complex64>
+
+var re = realf( z );
+// returns 10.0
+
+var im = imagf( z );
+// returns 12.0
+```
+
+Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.
+
+<!-- eslint-disable stdlib/capitalized-comments -->
+
+```javascript
+var Complex64Array = require( '@stdlib/array/complex64' );
+var Complex64 = require( '@stdlib/complex/float32' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+
+// Initial array...
```

Comment:
```suggestion
// Initial array:
```

`:` vs `...`: use colons to comment specific lines. Use ellipsis to comment multiple lines/blocks of code (e.g., for loops).